src/overlays/actors/ovl_Demo_Moonend/z_demo_moonend.c

```diff
@@ -28,28 +33,187 @@ ActorInit Demo_Moonend_InitVars = {
     (ActorFunc)DemoMoonend_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06001214;
-extern UNK_TYPE D_0600B540;
-extern UNK_TYPE D_060129F0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17B50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17B60.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17C48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17E70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17EE0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17FCC.s")
+void DemoMoonend_Init(Actor* thisx, PlayState* play) {
+    DemoMoonend* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.1f);
+    this->actionFunc = DemoMoonend_Idle;
+
+    if (DEMOMOONEND_GET_PARAMS_F(thisx) == 1) {
+        Actor_SetScale(&this->actor, 0.05f);
+        Actor_SetScale(&this->actor, 6.0f);
+        this->actor.draw = NULL;
+        this->cueType = CS_CMD_ACTOR_CUE_558;
+        this->actionFunc = func_80C17B60;
+    } else {
+        Actor_SetScale(&this->actor, 0.095f);
+        func_80183430(&this->skeletonInfo, &object_moonend_Blob_00B5A0, &object_moonend_Blob_001214, this->jointTable,
+                      this->morphTable, NULL);
+        func_801834A8(&this->skeletonInfo, &object_moonend_Blob_001214);
+        this->cueType = CS_CMD_ACTOR_CUE_560;
+        this->actionFunc = func_80C17C48;
+        this->actor.home.rot.z = 0;
+        this->actor.draw = NULL;
+        this->skeletonInfo.frameCtrl.unk_C = 2.0f / 3.0f;
+    }
+}
+
+void DemoMoonend_Destroy(Actor* thisx, PlayState* play) {
+    DemoMoonend* this = THIS;
+
+    if (DEMOMOONEND_GET_PARAMS_F(thisx) != 1) {
+        func_8018349C(&this->skeletonInfo);
+    }
+}
+
+void DemoMoonend_Idle(DemoMoonend* this, PlayState* play) {
```

Comment:
```suggestion
void DemoMoonend_DoNothing(DemoMoonend* this, PlayState* play) {
```
Pretty minor but we usually name these empty action functions as `DoNothing`.